airflow/models/dagrun.py

```diff
@@ -96,7 +94,7 @@ class DagRun(Base, LoggingMixin):
     last_scheduling_decision = Column(UtcDateTime)
     dag_hash = Column(String(32))
 
-    dag = None
+    dag: "Optional[DAG]" = None
```

Comment:
This is what is causing the docs build to fail.

Reply:
I have a fix for it -- so we can merge this and spend a bit more time looking at the sphinx upgrade.